python/tests/transfer_trigger_tests.py

```diff
@@ -0,0 +1,212 @@
+
+
+class Customer(object):
+    pass
+
+
+####################################################
+################### Triggers Here ##################
+####################################################
+
+
+def create_triggers():
+
+    @trigger(Customer, TriggerTime.before, TriggerAction.create, 1)
+    def before_create(dataframe, new, old, current):
+        print("[BEFORE create] - Procedure Executed")
+
+    @trigger(Customer, TriggerTime.after, TriggerAction.create, 1)
+    def after_create(dataframe, new, old, current):
+        print("[AFTER create] -- Procedure Executed")
+
+    @trigger(Customer, TriggerTime.before, TriggerAction.read, 1)
+    def before_read(dataframe, new, old, current):
+        print("[BEFORE READ] --- Procedure Executed")
+
+    @trigger(Customer, TriggerTime.after, TriggerAction.read, 1)
+    def after_read(dataframe, new, old, current):
+        print("[AFTER READ] ---- Procedure Executed")
+
+    @trigger(Customer, TriggerTime.before, TriggerAction.update, 1)
+    def before_update(dataframe, new, old, current):
+        print("[BEFORE UPDATE] - Procedure Executed")
+
+    @trigger(Customer, TriggerTime.after, TriggerAction.update, 1)
+    def after_update(dataframe, new, old, current):
+        print("[AFTER UPDATE] -- Procedure Executed")
+
+    @trigger(Customer, TriggerTime.before, TriggerAction.delete, 1)
+    def before_delete(dataframe, new, old, current):
+        print("[BEFORE DELETE] - Procedure Executed")
+
+    @trigger(Customer, TriggerTime.after, TriggerAction.delete, 1)
+    def after_delete(dataframe, new, old, current):
+        print("[AFTER DELETE] -- Procedure Executed")
+
+    return before_create, after_create, before_read, after_read, \
+        before_update, after_update, before_delete, after_delete
+
+def trigger_kinds():
+    return ["bc", "ac", "br", "ar", "bu",
+            "au", "bd", "bd"]
+
+def trigger_kinds_seprate():
+    return [(TriggerTime.before, TriggerAction.create), (TriggerTime.after, TriggerAction.create), (TriggerTime.before, TriggerAction.read),
+            (TriggerTime.after, TriggerAction.read), (TriggerTime.before, TriggerAction.update), (TriggerTime.after, TriggerAction.update),
+            (TriggerTime.before, TriggerAction.delete),(TriggerTime.after, TriggerAction.delete)]
+
+
+####################################################
+############## Unit Tests Start Here ###############
+####################################################
+
+class Test_trigger_transfer_test(unittest.TestCase):
+
+    def test_trigger_types(self):
+        """ Test if each kind of trigger has the correct attributes """
+
+        # Create an instance of every possible type of trigger
+        before_create, after_create, before_read, after_read, before_update, \
+            after_update, before_delete, after_delete = create_triggers()
+
+        # Perfrom tests that chech type, time, action, and priority
+        self.assertTrue(before_create.pcc_type == Customer)
```

Comment:
Make all of these in the form

self.assertEqual(Customer, before_create.pcc_type)

Reply:
Little tedious I know, but its better because the test failure messages are little more readable.